File: Data Structures & Algorithms/validate-parentheses/submission-1.java
class Solution {
    public boolean isValid(String s) {
        int l = s.length();
        Stack st = new Stack();
        for(int i=0;i<l;i++){
            char ele = s.charAt(i);

            if(ele == '(' || ele == '{' || ele == '['){
                st.push(ele);
            } else {
                if(ele == ')'){
                    if(!st.empty() && (char)st.peek() == '('){

                        st.pop();
                    } else {
                        return false ;
                    }
                } else if(ele == '}'){
                    if(!st.empty() && (char) st.peek() == '{'){
                        st.pop();
                    } else {
                        return false ;
                    }
                } else if(ele == ']'){
                    if(!st.empty() && (char)st.peek() == '['){
                        st.pop() ;
                    } else {
                        return false ;
                    }
                }
            }
        }
        return st.empty() ;
    }
}

File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-1.kt
class Solution {
    fun appendCharacters(s: String, t: String): Int {

        var ps = 0
        var pt = 0

        while(ps < s.length){
            if(pt < t.length){
                if(s[ps]==t[pt]){
                    ps++
                    pt++
                } else{
                    ps++
                }
            } else {
                ps++
            }
        }

        return t.length - pt
    }
}
